src/DockerImage.groovy

```diff
@@ -33,5 +33,16 @@ public class DockerImage implements Serializable {
         this.steps.sh "docker tag ${repository}:${tag} ${repository}:${newTag}"
         return new DockerImage(steps, repository, newTag)
     }
+
+    public DockerImage renameAndTag(newRepository, newTag)
+    {
+        this.steps.sh "docker tag ${repository}:${tag} ${newRepository}:${newTag}"
+        return new DockerImage(steps, newRepository, newTag)
+    }
+
+    public DockerImage tagWithSuffix(newTag, suffix)
```

Comment:
naming is hard

Reply:
try running `./gradlew` which should download dependencies and run tests for this class

Reply:
Your comment makes me think I've broken it but Jenkins seem to be running the tests. The existing tests are not too comprehensive beyond compiling. I guess the passed in steps could be a mock. Would be good if there was a docker build for this.

Reply:
Shouldn't have broken anything because as you have seen the build runs them. Looking at the output, existing tests for `DockerCompose` work well because they can are purely unit tests, while stubbing the `sh` Jenkins step is an headache.

Reply:
Would change this signature for consistency with `renameAndTag` to:
```suggestion
    public DockerImage suffixAndTag(suffix, newTag)
```

Reply:
> (but they do not cover this class)

Output seems to include `TestDockerImage > should_use_a_class PASSED`, do I misinterpret something?



> Would change this signature for consistency with `renameAndTag` to:

How about `addSuffixAndTag(suffix, newTag)`?

Reply:
also good :+1:

Reply:
> Output seems to include TestDockerImage > should_use_a_class

Yes, but it's only creating an instance of the class without calling any method; poor coverage due to the dependencies.

Reply:
> Yes, but it's only creating an instance of the class without calling any method; poor coverage due to the dependencies.

Okay, I understand. Just not sure whether that's just a statement or whether you are asking me to do anything about it. Given you mentioned that it's painful and not my main objective I'm hesitant :-)